src/python/pants/jvm/package/deploy_jar.py

```diff
@@ -49,6 +49,7 @@ class DeployJarFieldSet(PackageFieldSet, RunFieldSet):
         JvmJdkField,
         Dependencies,
     )
+    run_in_sandbox_behavior = RunInSandboxBehavior.RUN_REQUEST_HERMETIC
```

Comment:
I keep thinking of any way we could mechanically enforce this, but come up reasonably dry...

Maybe we add tests for our own code and hope plugin authors do the same?

Reply:
There really isn't a good way to test this beyond the rule author doing the right thing. I found a bunch of places where we were using `@rule` instead of `@_uncacheable_rule` while refactoring the environments.

The problem mostly stems from it being difficult to notice when problematic non-Pants APIs are used (e.g. `os.environ`, `subprocess`, etc)